earthfile2llb/converter.go

```diff
@@ -1718,7 +1718,13 @@ func (c *Converter) ExpandWildcard(ctx context.Context, fullTargetName string, c
 	if c.target.IsRemote() {
 		target = c.target
 	} else {
-		target = parsedTarget
+		// For local targets, we need to determine the full path relative to the
+		// working directory of Earthly in order to glob for matching paths.
+		ref, err := domain.JoinReferences(c.target, parsedTarget)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to join references")
+		}
+		target = ref.(domain.Target)
 	}
 
 	matches, err := c.opt.Resolver.ExpandWildcard(ctx, c.opt.GwClient, c.platr, target, parsedTarget.GetLocalPath())
```

Comment:
I was wondering if `parsedTarget.GetLocalPath()` should be changed to `target.GetLocalPath()`?

Additionally, when I looked up the function which is being called which looks like:

```
// ExpandWildcard will expand a wildcard BUILD target in a local path or remote
// Git repository. The pattern is the path relative to the target path and
// should be in the form 'my/path/*'
func (r *Resolver) ExpandWildcard(ctx context.Context, gwClient gwclient.Client, platr *platutil.Resolver, target domain.Target, pattern string) ([]string, error) {
```

and it's expecting a `pattern` rather than a path here; the difference appears that is it's missing a `/*` suffix?

Reply:
The arguments have changed a bit. Let me see if I can clean this up a bit by moving the `IsRemote` checks into `ExpandWildcard`. It's a little confusing at the moment.

Reply:
Updated with some improvements. The remote/local handling is now fully abstracted, which is nice.

Reply:
nice change, definitely easier to follow.